pkg/registration/spoke/registration/creating_controller.go

```diff
@@ -148,3 +170,46 @@ func ClientConfigDecorator(externalServerURLs []string, caBundle []byte) Managed
 		return cluster
 	}
 }
+
+// equalClientConfigs compares two ClientConfig slices for equality
+func equalClientConfigs(configs1, configs2 []clusterv1.ClientConfig) bool {
+	if len(configs1) != len(configs2) {
+		return false
+	}
+	for i, config1 := range configs1 {
+		config2 := configs2[i]
+		if config1.URL != config2.URL {
+			return false
+		}
+		if !bytes.Equal(config1.CABundle, config2.CABundle) {
+			return false
+		}
+	}
+	return true
+}
+
+// equalLabels compares two label maps for equality
+func equalLabels(labels1, labels2 map[string]string) bool {
```

Comment:
we cannot do such equal since the label of cluster can also be set by user on hub. So such label setting can only happen when creating the cluster.

If the label exists already on the cluster, the controller cannot update the label or annotation since it might be updated by user on purpose.